tools/split-tetragon-gotests/main.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/cilium/tetragon/pkg/vmtests"
+)
+
+var (
+	tetragonDir = "."
+	goTestsDir  = filepath.Join(tetragonDir, "go-tests")
+	outDir      = filepath.Join(tetragonDir, "testing")
+)
+
+func splitProgs(n int, blacklist []vmtests.GoTest) error {
+	tests, err := vmtests.ListTests(goTestsDir, false, blacklist)
+	if err != nil || len(tests) == 0 {
+		return fmt.Errorf("no tests found (err:%w). Is %s accesible? Did you run `make test-compile`?", err, goTestsDir)
+	}
+
+	files := make([]*os.File, n)
+	for i := 0; i < n; i++ {
+		fname := filepath.Join(outDir, fmt.Sprintf("test-group-%d", i))
+		files[i], err = os.Create(fname)
+		if err != nil {
+			return err
+		}
+		defer files[i].Close()
+	}
+
+	// NB: Some programs take longer than others. We allocate them
+	// one-by-one in groups hoping that this will lead to a decent
+	// load-balance between the groups.
+	i := 0
+	for _, test := range tests {
+		fmt.Fprintf(files[i%n], "%s\n", test.ToString())
+		i++
+	}
+
+	return nil
+}
+
+var CiBlacklist = []vmtests.GoTest{
+	// pkg.exporter has a rate limit test, which is time-dependent. There
+	// was a previous attempt to fix the test, but failed. Ignore it for
+	// now.
+	{PackageProg: "pkg.exporter"},
+	// https://github.com/cilium/tetragon/issues/247
+	{PackageProg: "pkg.sensors.tracing", Test: "TestCopyFd"},
+}
+
+func usage() {
+	fmt.Fprintf(
+		flag.CommandLine.Output(),
+		"Usage: %s [flags] <ngroups> -- split tests in <ngroups> groups. Output is %s/test-group-X.\n",
+		os.Args[0], outDir,
+	)
+	flag.PrintDefaults()
+}
+
+func main() {
+
+	var cirun bool
+	flag.BoolVar(&cirun, "ci-run", false, "This option enables CI blacklist for tests")
```

Comment:
Could we add a flag here for manually specifying skipped tests too? I'm specifically thinking of splitting e2e-framework into a separate job when that lands for example.

Reply:
I think that makes sense. How about we do it as a followup when it is needed?